rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/directives/AnyParamDirectives.scala

```diff
@@ -3,31 +3,94 @@ package org.elasticmq.rest.sqs.directives
 import akka.http.scaladsl.model.{FormData, HttpRequest}
 import akka.http.scaladsl.server.{Directives, Route, UnsupportedRequestContentTypeRejection}
 import akka.stream.Materializer
+import org.elasticmq.rest.sqs.{AWSProtocol, ContextPathModule, QueueURLModule}
+import org.elasticmq.rest.sqs.Constants.{QueueNameParameter, QueueUrlParameter}
+import org.elasticmq.rest.sqs.directives.QueueDirectives.AccountIdRegex
+import org.elasticmq.rest.sqs.model.{JsonData, RequestPayload}
 
 trait AnyParamDirectives {
-  this: Directives =>
+  this: Directives with ContextPathModule with QueueURLModule =>
 
-  private def entityOrEmpty =
+  private def formDataOrEmpty =
     entity(as[FormData]).recoverPF {
       // #68: some clients don't set the body as application/x-www-form-urlencoded, e.g. perl
       case Seq(UnsupportedRequestContentTypeRejection(_)) =>
         provide(FormData.Empty)
     }
 
-  private def extractAwsXRayTracingHeader(request: HttpRequest): Map[String, String] = {
+  private val actionHeaderPattern = """^AmazonSQS\.(\w+)$""".r
+  private def extractActionFromHeader(request: HttpRequest) =
+    request.headers
+      .find(_.name() == "X-Amz-Target")
+      .flatMap(
+        _.value() match {
+          case actionHeaderPattern(action) => Some(action)
+          case _                           => None
+        }
+      )
+      .getOrElse(throw new IllegalArgumentException("Couldn't find header X-Amz-Target"))
+  private def extractAwsXRayTracingHeader(request: HttpRequest): Option[String] = {
     request.headers
       .find(_.name().equalsIgnoreCase("X-Amzn-Trace-Id"))
-      .map(header => header.name() -> header.value())
-      .toMap
+      .map(_.value())
   }
 
-  def anyParamsMap(body: Map[String, String] => Route) = {
-    parameterMap { queryParameters =>
-      extractRequest { request =>
-        entityOrEmpty { fd =>
-          body((fd.fields.toMap ++ queryParameters ++ extractAwsXRayTracingHeader(request)).filter(_._1 != ""))
+  private def extractQueueNameAndUrlFromRequest(body: Map[String, String] => Route): Route = {
```

Comment:
[minor] this looks like a `Directive` ;)

Reply:
Do you mean I should move it somewhere?

Reply:
no, the method should be a directive. Sth like `extractQueueNameAndUrlFromRequest(Map): Directive0`, no?